core/src/main/java/org/apache/accumulo/core/client/Connector.java

```diff
@@ -313,6 +313,23 @@ public abstract ConditionalWriter createConditionalWriter(String tableName,
    */
   public abstract ReplicationOperations replicationOperations();
 
+  /**
+   * @return {@link ConnectionInfo} which contains information about Connection to Accumulo
+   * @since 2.0.0
+   */
+  public abstract ConnectionInfo info();
+
+  /**
+   * Creates new Connector for new user with principal and token
+   *
+   * @param principal User name/principal
+   * @param token Authentication token
+   *
+   * @return Connector for new user
+   * @since 2.0.0
+   */
+  public abstract Connector createConnector(String principal, AuthenticationToken token) throws AccumuloSecurityException, AccumuloException;
```

Comment:
Another possible way to accomplish this is via the builder.

```java
  Connector existing = ...
  Connector newC = Connector.builder().usingConnectionInfo(existing.info()).usingPassword(user,pass).build();
```

Reply:
Update PR to do this 368c222c543eba07